Makefile

```diff
@@ -1,96 +1,84 @@
-#	docker save -o dist/flannel-$(TAG)-$(ARCH).docker $(REGISTRY):$(TAG)-$(ARCH)
 
 vendor:
 	go mod tidy
 	go mod vendor
+
+build_all_docker: vendor 
+	docker build \
+		--no-cache \
+		--build-arg GOLANG_VERSION=$(GOLANG_VERSION) \
+		--build-arg TAG=$(TAG) \
+		--build-arg ALL_ARCH=$(ALL_ARCH) \
+		--tag $(REGISTRY):$(TAG) \
+		--tag $(REGISTRY):$(TAG)-$(ARCH) \
+		-f Dockerfile \
+		.
+
 build_linux: vendor
 	GOOS=linux GOARCH=$(ARCH) scripts/build_flannel.sh
 
 build_windows: vendor
-	GOOS=windows scripts/build_flannel.sh
-
-# This will build flannel cni-plugin natively using golang image
-dist/flannel-$(ARCH): dist/qemu-$(ARCH)-static
-	# valid values for ARCH are [amd64 arm arm64 ppc64le s390x]
-	docker run -e CGO_ENABLED=$(CGO_ENABLED) -e GOARCH=$(ARCH) -e GOCACHE=/go -e GOOS=linux -e  GOFLAGS="${GOFLAGS} -mod=vendor" \
-		-u $(shell id -u):$(shell id -g) \
-		-v $(CURDIR)/dist/qemu-$(ARCH)-static:/usr/bin/qemu-$(ARCH)-static \
-		-v $(CURDIR):/go/src/github.com/flannel-io/flannel:ro \
-		-v $(CURDIR)/bin:/go/src/github.com/flannel-io/flannel/bin \
-		-v $(CURDIR)/dist:/go/src/github.com/flannel-io/flannel/dist \
-		golang:$(GO_VERSION) /bin/bash -c '\
-		cd /go/src/github.com/flannel-io/flannel && \
-        go build -o "./bin/flannel" . && \
-		mv bin/flannel dist/flannel-$(ARCH)'
-
-dist/qemu-%-static:
-	if [ "$(@F)" = "qemu-amd64-static" ]; then \
-		wget -O dist/qemu-amd64-static https://github.com/multiarch/qemu-user-static/releases/download/$(QEMU_VERSION)/qemu-x86_64-static; \
-	elif [ "$(@F)" = "qemu-arm64-static" ]; then \
-		wget -O dist/qemu-arm64-static https://github.com/multiarch/qemu-user-static/releases/download/$(QEMU_VERSION)/qemu-aarch64-static; \
-	else \
-		wget -O dist/$(@F) https://github.com/multiarch/qemu-user-static/releases/download/$(QEMU_VERSION)/$(@F); \
-	fi 
-
-docker-push: dist/flannel-$(TAG)-$(ARCH).docker
-	docker push $(REGISTRY):$(TAG)-$(ARCH)
-
-docker-manifest-amend:
-	DOCKER_CLI_EXPERIMENTAL=enabled docker manifest create --amend $(REGISTRY):$(TAG) $(REGISTRY):$(TAG)-$(ARCH)
-
-docker-manifest-push:
-	DOCKER_CLI_EXPERIMENTAL=enabled docker manifest push --purge $(REGISTRY):$(TAG)
-
-docker-push-all:
-	ARCH=amd64 make docker-push docker-manifest-amend
-	ARCH=arm make docker-push docker-manifest-amend
-	ARCH=arm64 make docker-push docker-manifest-amend
-	ARCH=ppc64le make docker-push docker-manifest-amend
-	ARCH=s390x make docker-push docker-manifest-amend
-	make docker-manifest-push
-
-test_linux:
-	scripts/test_linux.sh
-
-build_image: build_linux
-	docker build . -f Dockerfile.linux
-
-push_image: build_image
-	docker push quay.io/coreos/flannel-cni:${VERSION}
+	GOOS=windows GOARCH=$(ARCH) scripts/build_flannel.sh
+
+test_linux: vendor build_linux
+	TAG=$(TEST_TAG) scripts/test_linux.sh
 
 clean:
 	rm -f dist/flannel*
-	rm -f dist/qemu-*
+	rm -f release/cni-plugin-flannel*
 
-#release: build_image test_linux push_image
+package: vendor
```

Comment:
Don't we need the binaries to be present when packaging? In that case, we would need to add `build_all` as a prerequisite or?

Reply:
I explained it more in-depth [here](https://github.com/flannel-io/cni-plugin/pull/22#discussion_r777589532) but the package target is meant to only be ran as part of the full automation or used as a dev/debug/qa aid. 

This came in handy when I was writing and testing the scripts/builds as it greatly reduced the compute time required to test the packaging portion of the release process.

Reply:
ok